### PR TITLE
Adjusting NearProvider to EVM Precompile

### PR DESCRIPTION
**Pull request by ilblackdragon**
**Unit tests completed?:** _Y_

**Relevant README/Documentation Updated?:** _N_

### CODE REVIEW

**Comment by ilblackdragon** [comment]
Just to leave it here: When an account from key store doesn't exist on chain - it doesn't really fail properly.

**Review by mikedotexe — APPROVED** [review]

**Comment by mikedotexe** [comment]
We will wait until this PR is merged and a release is published to npm, then we will modify the dependency for `near-api-js` accordingly:
https://github.com/near/near-api-js/pull/388

File: test/key_management.test.js
Comment (mikedotexe) [inline test/key_management.test.js:41]: We're getting to a good place with this branch, and I'm working on the failing tests. I've checked out the commit (8f061e54e80a73a6346e829f54ae4adb5402ba5f) where this was entered and see it was never passing. I hate to just change this value to what the tests say is the correct signature, but I think that's on the table as this one slipped by my review. If there's reason to think this test is supposed to pass, please let me know.

Comment (mikedotexe) [reply test/key_management.test.js]: Figured it out and changes a few tests, resolving.

**Comment by mikedotexe** [comment]
There are three tests failing, but I don't think these are blockers to merging. I'll be talking more closely with other EVM folks about them.
One test in the near-web3-provider that seems to sometimes have different values:

```
---- CONTRACT INTERACTION ---- › sendTransaction | eth_sendTransaction › sends the correct balance when simply transferring funds to other evm address
    expect(received).toStrictEqual(expected) // deep equality
    Expected: 4.141e+21
    Received: 4.1410000000000003e+21
```

two other tests failing on the near-web3-provider besides the one I mentioned above:

```
sends near back to nearAccount if sufficient funds in corresponding evm account
    expect(received).toStrictEqual(expected) // deep equality
    Expected: 3000000000000000000
    Received: 3000000000000262000
```

and

```
forces capitalized near accountID to lowercase and successfully completes transaction
    expect(received).toStrictEqual(expected) // deep equality
    Expected: 3000000000000000000
    Received: 2999999999999738000
```

**Pull request merged**
